core-chart/templates/postcreatehooks/its-hub-init.yaml

```diff
@@ -0,0 +1,50 @@
+{{- if .Values.InstallPCHs }}
+apiVersion: tenancy.kflex.kubestellar.org/v1alpha1
+kind: PostCreateHook
+metadata:
+  name: its-hub-init
+  labels:
+    kflex.kubestellar.io/cptype: its
+spec:
+  templates:
+  - apiVersion: batch/v1
+    kind: Job
+    metadata:
+      name: "{{"{{.HookName}}"}}"
+    spec:
+      template:
+        spec:
+          initContainers:
+          - name: wait-for-config-incluster
+            image: quay.io/kubestellar/kubectl:{{.Values.KUBECTL_VERSION}}
+            command:
+              - sh
+              - -c
+              - |
+                echo "Waiting for config-incluster in Secret {{.ITSSecretName}}..."
+                until kubectl get secret "{{.ITSSecretName}}" -n {{.ControlPlaneName}}-system -o jsonpath="{.data.config-incluster}" | grep -q .; do
```

Comment:
There is good news and bad news for you in https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod:

> When a volume contains data from a Secret, and that Secret is updated, Kubernetes tracks this and updates the data in the volume, using an eventually-consistent approach.

What this initContainer needs to accomplish is to ensure that when the regular container tries to use `/etc/kube/{{ .ITSkubeconfig }}`, that file exists. Because the kubelet's update of the volume is only eventually consistent with the Secret, waiting for the data member to exist in the Secret is not enough.

The good news is that it is enough to wait for the file to appear.